File: app/src/main/java/com/paradoxo/amadeus/activity/SobreActivity.java
package com.paradoxo.amadeus.activity;

import android.content.Intent;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.google.android.gms.oss.licenses.OssLicensesMenuActivity;
import com.paradoxo.amadeus.R;

public class SobreActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_sobre);

        final String linkPlaylistAmadeusYouTube = getString(R.string.link_playlist_amadeus_yt);

        try {
            PackageInfo pInfo = getPackageManager().getPackageInfo(getPackageName(), 0);
            String versao = getString(R.string.versao) + " " + pInfo.versionName;

            TextView textView = findViewById(R.id.versaoTextView);
            textView.setText(versao);

        } catch (PackageManager.NameNotFoundException e) {
            e.printStackTrace();
        }


        (findViewById(R.id.irParaYouTubeButton)).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                Intent activityAmadeusYouTube = new Intent(Intent.ACTION_VIEW);
                activityAmadeusYouTube.setData(Uri.parse(linkPlaylistAmadeusYouTube));
                startActivity(activityAmadeusYouTube);

            }
        });

        (findViewById(R.id.irParaYouTubeButton)).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                Intent activityAmadeusYouTube = new Intent(Intent.ACTION_VIEW);
                activityAmadeusYouTube.setData(Uri.parse(linkPlaylistAmadeusYouTube));
                startActivity(activityAmadeusYouTube);

            }
        });

        findViewById(R.id.licencaCodigoTextView).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(SobreActivity.this, OssLicensesMenuActivity.class));

            }
        });

        findViewById(R.id.lincencaAnimacoesTextView).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intentActivtyAnimcaoes = new Intent(SobreActivity.this, LicencaAnimacoesActivity.class);
                startActivity(intentActivtyAnimcaoes);
            }
        });
    }

}

File: app/src/main/java/com/paradoxo/amadeus/activity/SobreActivity.kt
package com.paradoxo.amadeus.activity

import android.content.Intent
import android.content.pm.PackageManager
import android.net.Uri
import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.google.android.gms.oss.licenses.OssLicensesMenuActivity
import com.paradoxo.amadeus.R
import com.paradoxo.amadeus.util.Util.configurarToolBarBranca

class SobreActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_sobre)
        configurarInterface()
    }

    private fun configurarInterface() {
        configurarToolBarBranca(this)
        carregarTextoVersao()
        configurarClickBotoes()
        configurarClikLicencas()
    }

    private fun configurarClickBotoes() {
        configurarClickLink(R.id.irParaYouTubeLayout, getString(R.string.link_playlist_amadeus_yt))
        configurarClickLink(R.id.enviarFeedBackLayout, getString(R.string.linkAmadeusGooglePlayStore))
        configurarClickLink(R.id.gitHubLayout, getString(R.string.linkAmadeusGitHub))
        configurarClickLink(R.id.creditosLottieApi1TextView, getString(R.string.link_lottie_api_1))
        configurarClickLink(R.id.creditosLottieApi2TextView, getString(R.string.link_lottie_api_2))
    }

    private fun carregarTextoVersao() {
        try {
            val pInfo = packageManager.getPackageInfo(packageName, 0)
            val versao = getString(R.string.versao) + " " + pInfo.versionName
            findViewById<TextView>(R.id.versaoTextView).text = versao
        } catch (e: PackageManager.NameNotFoundException) {
            e.printStackTrace()
        }
    }

    private fun configurarClickLink(viewSelecionada: Int, link: String) {
        findViewById<android.view.View>(viewSelecionada).setOnClickListener {
            val intent = Intent(Intent.ACTION_VIEW)
            intent.data = Uri.parse(link)
            startActivity(intent)
        }
    }

    private fun configurarClikLicencas() {
        findViewById<android.view.View>(R.id.licencaCodigoLayout).setOnClickListener {
            startActivity(Intent(this@SobreActivity, OssLicensesMenuActivity::class.java))
        }
    }
}
